tests/conv2d_model.h

```diff
@@ -0,0 +1,204 @@
+#if RTNEURAL_USE_EIGEN
+
+#pragma once
+
+#include "load_csv.hpp"
+#include "util_tests.hpp"
+#include <RTNeural.h>
+
+/**
+ * Computes the receptive field of the model (always 1 if there's no Conv2D layer in the model)
+ * Also corresponds to the number of run after a reset before getting a valid/meaningful output
+ */
+int computeReceptiveField(const RTNeural::Model<TestType>& model)
+{
+    int receptive_field = 1;
+
+    for(auto* l : model.layers)
+    {
+        if(l->getName() == "conv2d")
+        {
+            auto conv = dynamic_cast<RTNeural::Conv2D<TestType>*>(l);
+            receptive_field += conv->receptive_field - 1;
+        }
+    }
+    return receptive_field;
+}
+
+int computeTotalPaddedLeftFramesTensorflow(const RTNeural::Model<TestType>& model)
+{
+    int total_pad_left = 0;
+
+    for(auto* l : model.layers)
+    {
+        if(l->getName() == "conv2d")
+        {
+            // Following tensorflow padding documentation: https://www.tensorflow.org/api_docs/python/tf/nn#notes_on_padding_2
+            // And using the fact that in the time dimension, only stride = 1 is supported:
+
+            auto conv = dynamic_cast<RTNeural::Conv2D<TestType>*>(l);
+
+            if(!conv->valid_pad)
+                total_pad_left += (conv->receptive_field - 1) / 2;
+        }
+    }
+
+    return total_pad_left;
+}
+
+void processModelNonT(RTNeural::Model<TestType>& model, const std::vector<TestType>& xData, std::vector<TestType>& yData, int num_frames, int num_features_in, int num_features_out)
+{
+    model.reset();
+
+    TestType input alignas(RTNEURAL_DEFAULT_ALIGNMENT)[num_features_in];
```

Comment:
Use a vector here instead as `num_features_in` is not const (seems to be the reason for the crash on windows)